src/Uno.UI/UI/Xaml/Media/ImageSource.cs

```diff
@@ -98,16 +98,13 @@ internal void InitFromUri(Uri uri)
 			if (uri.IsLocalResource())
 			{
 				InitFromResource(uri);
-				return;
 			}
-
-			if (uri.IsAppData())
+			else if (uri.IsAppData())
 			{
 				var filePath = AppDataUriEvaluator.ToPath(uri);
 				InitFromFile(filePath);
 			}
-
-			if (uri.IsFile)
+			else if (uri.IsFile)
 			{
 				InitFromFile(uri.PathAndQuery);
 			}
```

Comment:
The real change is removing the "return" to let the uri be assigned to WebUri. Using `else if` just reduces unnecessary work since only one condition can be true.